samcli/lib/config/samconfig.py

```diff
@@ -148,24 +151,38 @@ def config_dir(template_file_path=None):
         return os.getcwd()
 
     def _read(self):
-        if self.document:
-            return self.document
-
-        try:
-            txt = self.filepath.read_text()
-            self.document = tomlkit.loads(txt)
-        except OSError:
-            self.document = tomlkit.document()
-
+        if not self.document:
+            try:
+                txt = self.filepath.read_text()
+                self.document = tomlkit.loads(txt)
+                self._version_sanity_check(self._version())
+            except OSError:
+                self.document = tomlkit.document()
+
+        if self.document.body:
+            self._version_sanity_check(self._version())
         return self.document
 
     def _write(self):
         if not self.document:
             return
         if not self.exists():
             open(self.filepath, "a+").close()
+        current_version = self._version() if self._version() else SAM_CONFIG_VERSION
+        try:
```

Comment:
Just curious: any reason why we are not checking for key to be present and add only if it isn't?

Reply:
https://github.com/awslabs/aws-sam-cli/pull/1581/files#r349833716

Reply:
we do.